### PR TITLE
DeviceMemoryUsage: add AMD support, use ProcessLogger for version.sh

### PR DESCRIPTION
**Pull request by matthiasdiener**
**Questions for the review**:
- [ ] Is the scope and purpose of the PR clear?
  - [ ] The PR should have a description.
  - [ ] The PR should have a guide if needed (e.g., an ordering).
- [ ] Is every top-level method and class documented? Are things that should be documented actually so?
- [ ] Is the interface understandable? (I.e. can someone figure out what stuff does?) Is it well-defined?
- [ ] Does the implementation do what the docstring claims?
- [ ] Is everything that is implemented covered by tests?
- [ ] Do you see any immediate risks or performance disadvantages with the design? Example: what do interface normals attach to?

### CODE REVIEW

File: mirgecom/array_context.py [hunk 1]
```diff
@@ -176,6 +180,10 @@ def _check_gpu_oversubscription(actx: ArrayContext) -> None:
     Only works with CUDA devices currently due to the use of the
     PCI_DOMAIN_ID_NV extension.
     """
+    import sys
+    if "mpi4py" not in sys.modules:
+        return
+
```
Comment (majosm) [inline mirgecom/array_context.py:186]: Would it make sense to add an `actx_class_is_distributed` or `actx_class_is_mpi` function and only call `_check_cache_dirs_node`/`_check_gpu_oversubscription` if it returns `True`?

File: mirgecom/array_context.py [hunk 2]
```diff
@@ -227,11 +235,16 @@ def _check_gpu_oversubscription(actx: ArrayContext) -> None:
...
 
     assert isinstance(actx, (PyOpenCLArrayContext, PytatoPyOpenCLArrayContext))
 
-    rank = MPI.COMM_WORLD.Get_rank()
+    if "mpi4py" in sys.modules:
+        from mpi4py import MPI
+        rank = MPI.COMM_WORLD.Get_rank()
+    else:
+        rank = 0
+
```
Comment (majosm) [inline mirgecom/array_context.py:247]: Can you check if it's an MPI array context and get `actx.mpi_communicator`?

**Review by majosm — APPROVED** [review]
Couple of minor suggestions, otherwise LGTM.

File: mirgecom/array_context.py (hunk 1 above)
Comment (matthiasdiener) [reply mirgecom/array_context.py]: Thanks! What do you think of [16a3a6a](https://github.com/illinois-ceesd/mirgecom/pull/1060/commits/16a3a6a9560db47c01b18bcac13ad366a10f8c04)?

File: mirgecom/array_context.py (hunk 2 above)
Comment (matthiasdiener) [reply mirgecom/array_context.py]: Done in [16a3a6a](https://github.com/illinois-ceesd/mirgecom/pull/1060/commits/16a3a6a9560db47c01b18bcac13ad366a10f8c04)

**Pull request merged**